ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/ontologies/presenter/OntologiesPresenterImpl.java

```diff
@@ -331,10 +360,16 @@ void helperMap(List<OntologyHierarchy> children) {
     @Override
     public void onSaveOntologyHierarchy(SaveOntologyHierarchyEvent event) {
         List<String> iris = event.getIris();
-
-        for (final String iri : iris) {
+        boolean clearHierarchies = true;
+        boolean isLast = false;
+        for (String iri : iris) {
+            if (iris.indexOf(iri) == iris.size() - 1) {
+                isLast = true;
+            }
+            view.maskHierarchyTree();
             serviceFacade.saveOntologyHierarchy(event.getOntology().getVersion(),
-                                                iri, new SaveHierarchyAsyncCallback(iri));
+                                                iri, new SaveHierarchyAsyncCallback(iri, clearHierarchies, isLast));
+            clearHierarchies = false;
```

Comment:
That's a really good point that I hadn't thought of 👍 
I'll switch to your method of refreshing after each success.